arduino-core/src/processing/app/debug/LegacyTargetPlatform.java

```diff
@@ -245,4 +245,9 @@ public String toString() {
       res += "  " + boardId + " = " + boards.get(boardId) + "\n";
     return res + "}";
   }
+
+  @Override
+  public boolean isInSketchbook() {
+	return getFolder().getAbsolutePath().startsWith(BaseNoGui.getDefaultSketchbookFolder().getAbsolutePath());
```

Comment:
You're using `getDefaultSketchbookFolder()` here, not the actual folder (which I customized through the preferences). I would even suggest using `getSketchbookHardwareFolder()` instead, which is more specific. I just tried this, which works as expected on my system.

```suggestion
	return getFolder().getAbsolutePath().startsWith(BaseNoGui.getSketchbookHardwareFolder().getAbsolutePath());
```

Reply:
:+1: